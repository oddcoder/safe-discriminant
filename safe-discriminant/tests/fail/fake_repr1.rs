// attempting to confuse Discriminant into thinking
// the fake repr is the real repr
use safe_discriminant::Discriminant;
use safe_discriminant_derive::repr;

#[derive(Discriminant)]
#[repr(u8)]
pub enum Foo {
    A = 0,
    B = 1,
}

fn main() {}
