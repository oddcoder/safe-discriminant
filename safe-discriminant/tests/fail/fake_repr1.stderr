error[E0659]: `repr` is ambiguous
 --> tests/fail/fake_repr1.rs:7:3
  |
7 | #[repr(u8)]
  |   ^^^^ ambiguous name
  |
  = note: ambiguous because of a name conflict with a builtin attribute
  = note: `repr` could refer to a built-in attribute
note: `repr` could also refer to the attribute macro imported here
 --> tests/fail/fake_repr1.rs:4:5
  |
4 | use safe_discriminant_derive::repr;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: use `crate::repr` to refer to this attribute macro unambiguously
